lib/test_state.go

```diff
@@ -0,0 +1,34 @@
+package lib
+
+import (
+	"io"
+
+	"github.com/sirupsen/logrus"
+	"go.k6.io/k6/metrics"
+)
+
+// TestPreInitState contains all of the state that can be gathered and built
+// before the test run is initialized.
+type TestPreInitState struct {
+	RuntimeOptions RuntimeOptions
+	Registry       *metrics.Registry
+	BuiltinMetrics *metrics.BuiltinMetrics
+	KeyLogger      io.WriteCloser
+
+	// TODO: replace with logrus.FieldLogger when all of the tests can be fixed
+	Logger *logrus.Logger
+}
+
+// TestRunState contains the pre-init state as well as all of the state and
+// options that are necessary for actually running the test.
+type TestRunState struct {
```

Comment:
```suggestion
type TestRun struct {
```
Do we really need the word `State`? Every struct is a state of something in the end.

Reply:
hmm though the `Runner` and `Options` are not exactly "state"... :sweat_smile: :man_shrugging: 

:+1: from me for renaming it to just `TestRun` - I am fine with either name, what do the rest of the @grafana/k6-devs think?

Reply:
Personally, I find `TestRunState` more explicit. It's somewhat consistent with the rest of our naming, where we already have things like `ExecutionState` and `RuntimeOptions`, with a clear distinction between `State` (something is running, and this is where its knobs are at), and `Options` (whenever you need to decide what behavior to observe, or which state to go in, look at those values).